### PR TITLE
Create UT for ctor&dtor of all HMI&MOBILE commands

### PR DESCRIPTION
**Pull request by okozlovlux**
(both request and response):
  - created file dummy_mobile_commands_tests.cc with tests
    for mobile commands
  - created file dummy_hmi_commands_tests.cc with tests
    for HMI commands

Related to APPLINK-24964

@LevchenkoS @AByzhynar @Kozoriz @vlantonov @anosach-luxoft @VVeremjova @wolfylambova22 please review

### CODE REVIEW

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+    ON_CALL(app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(mock_event_dispatcher_));
+    ON_CALL(app_mngr_, get_settings()).WillByDefault(ReturnRef(app_mngr_set_));
+    ON_CALL(app_mngr_set_, start_stream_retry_amount())
+        .WillByDefault(ReturnRef(stream_retry_));
+  }
+
+ protected:
+  std::pair<uint32_t, int32_t> stream_retry_;
+  NiceMock<MockApplicationManagerSettings> app_mngr_set_;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @okozlovlux, `MockApplicationManagerSettings` already exist in `CommandsTest` as `app_mngr_settings_`.

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+  NiceMock<MockEventDispatcher> mock_event_dispatcher_;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @okozlovlux, `MockEventDispatcher` already included in `CommandRequestTest` class. Use it to deal with request commands.

File: src/components/application_manager/test/commands/mobile/dummy_mobile_commands_test.cc [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    ON_CALL(app_mngr_, get_settings()).WillByDefault(ReturnRef(app_mngr_set_));
+    ON_CALL(app_mngr_set_, app_icons_folder())
+        .WillByDefault(ReturnRef(kEmptyString_));
+  }
+
+ protected:
+  const std::string kEmptyString_ = "";
+
+  NiceMock<MockApplicationManagerSettings> app_mngr_set_;
+  NiceMock<MockEventDispatcher> mock_event_dispatcher_;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/dummy_mobile_commands_test.cc]: @okozlovlux, same notes as in `dummy_hmi_commands_test.cc`

**Comment by vlantonov** [comment]
Reviewed

**Comment by wolfylambova22** [comment]
Reviewed

**Comment by VVeremjova** [comment]
@okozlovlux Reviewed

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @LevchenkoS fixed 97e65a0

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc (hunk 2 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @LevchenkoS fixed 97e65a0

File: src/components/application_manager/test/commands/mobile/dummy_mobile_commands_test.cc (hunk 3 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/dummy_mobile_commands_test.cc]: @LevchenkoS fixed 97e65a0

**Comment by LevchenkoS** [comment]
@okozlovlux, reviewed.

**Comment by Kozoriz** [comment]
Reviewed

**Comment by AByzhynar** [comment]
@okozlovlux I think such information (`macro TYPED_TEST_CASE takes max 50 args. That is why there are few TYPED_TEST_CASE for HMI and mobile commands`)should be added as comment in file . As PR will be merged and branch deleted.

**Comment by okozlovlux** [comment]
@AByzhynar will be fixed in the next commit

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+};
+
+template <class Command>
+class HMICommandsTestThird : public HMICommandsTest<Command> {
+ public:
+  using typename HMICommandsTest<Command>::CommandType;
+};
+
+template <class Command>
+class HMICommandsTestFour : public HMICommandsTest<Command> {
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @okozlovlux `Fourth` and `Fifth` below

**Comment by anosach-luxoft** [comment]
Reviewed

**Comment by AByzhynar** [comment]
Reviewed

**Pull request merged**